commerce_coordinator/apps/iap/api/v1/views.py

```diff
@@ -0,0 +1,124 @@
+    """
+    API view for preparing a cart in CT and then converting it to an order
+    for mobile In-App purchase
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request: Request) -> Response:
+        """
+        Handles POST request for preparing a cart in CT and then converting it
+        to an order for mobile In-App purchase
+        """
+        try:
+            serializer = OrderRequestSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+            request_data: OrderRequestData = serializer.validated_data  # type: ignore
+
+            external_price = Money(
+                cent_amount=int(request_data["price"] * 100),
+                currency_code=request_data["currency"],
+            )
+            standalone_price = get_standalone_price_for_sku(
+                sku=request_data["course_run_key"]
+            )
+
+            client = CommercetoolsAPIClient()
+            customer = get_ct_customer(client, request.user)
+            cart = client.get_customer_cart(customer.id)
+
+            if cart:
+                client.delete_cart(cart)
+
+            order_number = client.get_new_order_number()
+            cart = client.create_cart(
+                customer=customer,
+                order_number=order_number,
+                currency=request_data["currency"],
+            )
+            payment = client.create_payment(
+                amount_planned=external_price,
+                customer_id=customer.id,
+                # TODO: finalize source of these
+                payment_method="Dummy Card",
+                payment_status="Dummy Success",
+                payment_processor=request_data["payment_processor"],
+                # TODO: fetch from purchase token
+                psp_payment_id="Dummy-" + str(uuid.uuid4()),
+                psp_transaction_id="Dummy-" + str(uuid.uuid4()),
+                usd_cent_amount=standalone_price.cent_amount,
+            )
+            cart = client.update_cart(
+                cart=cart,
+                external_price=external_price,
+                sku=request_data["course_run_key"],
+                email_domain=get_email_domain(cart.customer_email),
+                payment_id=payment.id,
+            )
+            order = client.create_order_from_cart(cart)
+            order = client.update_line_items_transition_state(
+                order_id=order.id,
+                order_version=order.version,
+                line_items=order.line_items,
+                from_state_id=order.line_items[0].state[0].state.id,
+                new_state_key=TwoUKeys.PENDING_FULFILMENT_STATE,
+                use_state_id=True,
+            )
+
+            serializer = OrderResponseSerializer(
+                data={
+                    "order_id": order.id,
+                    "order_number": order.order_number,
+                }
+            )
+            serializer.is_valid(raise_exception=True)
```

Comment:
we don't need a serializer here since order_id and order_number are already in string format so we can directly return data here